packages/backend/src/invoice/service.test.ts

```diff
@@ -0,0 +1,220 @@
+import Knex from 'knex'
+import { WorkerUtils, makeWorkerUtils } from 'graphile-worker'
+
+import { InvoiceService } from './service'
+import { createTestApp, TestContainer } from '../tests/app'
+import { Invoice } from './model'
+import { resetGraphileDb } from '../tests/graphileDb'
+import { GraphileProducer } from '../messaging/graphileProducer'
+import { Config } from '../config/app'
+import { IocContract } from '@adonisjs/fold'
+import { initIocContainer } from '../'
+import { AppServices } from '../app'
+import { User } from '../user/model'
+import { UserService } from '../user/service'
+import { truncateTable, truncateTables } from '../tests/tableManager'
+
+describe('Invoice Service', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let workerUtils: WorkerUtils
+  let invoiceService: InvoiceService
+  let userService: UserService
+  let user: User
+  let knex: Knex
+  const messageProducer = new GraphileProducer()
+  const mockMessageProducer = {
+    send: jest.fn()
+  }
+
+  beforeAll(
+    async (): Promise<void> => {
+      deps = await initIocContainer(Config)
+      deps.bind('messageProducer', async () => mockMessageProducer)
+      appContainer = await createTestApp(deps)
+      workerUtils = await makeWorkerUtils({
+        connectionString: appContainer.connectionUrl
+      })
+      await workerUtils.migrate()
+      messageProducer.setUtils(workerUtils)
+      knex = await deps.use('knex')
+    }
+  )
+
+  beforeEach(
+    async (): Promise<void> => {
+      invoiceService = await deps.use('invoiceService')
+      userService = await deps.use('userService')
+      user = await userService.create()
+    }
+  )
+
+  afterAll(
+    async (): Promise<void> => {
+      await appContainer.shutdown()
+      await workerUtils.release()
+      await resetGraphileDb(knex)
+      await truncateTables(knex)
```

Comment:
The tables are already rolled back and destroyed in the teardown. This is more to prevent conflicts when running multiple test suits against the same table - i.e. just ensuring each test suite has a clean table to work with.